src/nl/surf/eduhub_rio_mapper/link.clj

```diff
@@ -41,7 +41,7 @@
 (defn- sleutel-changer [id finder]
   (fn [element]
     (if (finder element)
-      (assoc-in element [2 1] id)
+      (when id (assoc-in element [2 1] id))
```

Comment:
This will return `nil` when `id` is falsy, is this what you mean of would you like to return the `element` in that case?  Using an `if` here is probably clearer..